[Gatsby Docs Update] Fixed iOS overscroll issues, and redesigned mobile sidebar button to …

joecritch opened the pull request:
- Added some fixes from http://blog.christoffer.online/2015-06-10-six-things-i-learnt-about-ios-rubberband-overflow-scrolling/ for the overflow scrolling. Seems to work.
- Subtle sidebar transition tweaks
- Redesigned mobile sidebar toggle. Previously there was an issue where you'd have to tap it twice, as only Safari's toolbar would appear on the first tap. ![screen shot 2017-09-20 at 18 38 49](https://user-images.githubusercontent.com/24449/30658874-0b8a7c3c-9e34-11e7-9f3e-54001d0a9522.png). This caused a bit of yak shaving on the redesign, but I think it works just as well, if not better.

@bvaughn

bvaughn commented on www/src/components/StickyResponsiveSidebar/StickyResponsiveSidebar.js:
```diff
@@ -50,12 +50,80 @@ class StickyResponsiveSidebar extends Component {
+    document.addEventListener('touchmove', this.handleDocumentTouchMove);
+    document
+      .querySelector('.scrollable')
+      .addEventListener('touchstart', this.handleSidebarTouchStart);
```
Strong initial dislike to all of the global handlers/selectors here. If we have to do this, I think we should wrap it up in a separate component rather than mix it into the sticky component.

flarnie replied:
It might be possible to make it a bit more 'React' ish. For now we could start by adding a comment and linking to the blog post that describes this fix for the iOS overscroll bug. (http://blog.christoffer.online/2015-06-10-six-things-i-learnt-about-ios-rubberband-overflow-scrolling/)

bvaughn commented on www/src/components/LayoutHeader/Header.js:
```diff
@@ -23,8 +23,19 @@ import logoSvg from 'icons/logo.svg';
+
+css.global('.is-menu-open #header', {
+  transform: 'translateY(-100%)',
+});
```
It would be nice to avoid global styles for this. I suggest we pass an `isOpen` prop do this component instead and (probably) expose a toggle function to the sticky sidebar via context.

bvaughn commented on www/src/components/Flex/Flex.js:
```diff
@@ -25,11 +25,13 @@ const Flex = ({
   shrink = 1,
   type = 'div',
   valign = 'flex-start',
+  className,
```
nit: Alpha-sort props 😄

joecritch commented:
@bvaughn @flarnie 

- Fixed the scrolling issue. It was actually the `overflow-x: hidden` on the article. I've fixed the horizontal scrolling with a better solution. 
- Agreed that the text truncation is a trade-off with the floating button. However, you do already have context of what page you're on, and I feel like it's better than a full-width button when being 44px further up.
- Yeah, the global event handlers and CSS aren't great. Don't have the time to improve these tonight, so any help would be great.

Mathspy commented:
For the lack of a better place and for the sake of not to opening my first React PR to add a line:

This tiny little visual bug (the unneeded scroll bar) exists on Chrome 61, and it should be fixed with a little ``overflow-y: hidden``
![Tiny Little Bug](https://i.imgur.com/FnE60wb.png)

Loving the new website so far ❤️ ❤️ ❤️ Amazing work everyone!

joecritch commented:
@Mathspy Thank you! It's on the list 👍

joecritch commented on www/src/templates/docs.js:
```diff
@@ -30,7 +30,7 @@ const sectionList = sectionListA
   );
 
-const Docs = ({data, location}) => (
+const Docs = ({data, location}) => console.log('render Docs') || (
```
this doesn't fire when you resize the window (above xsmall)

flarnie commented:
Just manually tested on Android 7.0, Samsung Galaxy S6. Looks and feels pretty good!

It seems like there is sometimes a lag where I can't scroll for a moment right after the page loads, but hard to say for sure. Still seems usable to me.

bvaughn commented:
> It seems like there is sometimes a lag where I can't scroll for a moment right after the page loads, but hard to say for sure. Still seems usable to me.

Does this happen on master too, by chance?

bvaughn commented:
Okay! Taking another look at this 😁 

Trying what I think is the latest build of this ([59c43b440752d04852a6a86c](https://59c43b440752d04852a6a86c--reactjs.netlify.com/)) on my Android:
* The nav doesn't open when touched. Instead a Chrome pop expands from the bottom to show a link to more info about the tapped text (eg "Hello World"). If I'm careful and click on the arrows instead, this doesn't happen, but neither does the menu open.
* The text truncation still happens too. eg on the Tutorial page, the nav just shows "Tutorial: Int...".

flarnie commented:
Ok - so I wonder if the difference between Brian and my test is the version of mobile Chrome we are running?

Chrome 61 just came out, and had some breaking changes related to scrolling. We just fixed a bug in Draft.js related to this.

@bvaughn what version of mobile chrome do you have on your Android phone?

Mine is `60.0.3112.116`.

props to @sophiebits for talking through this and giving me the idea.

joecritch commented:
Thanks for the reviews, folks. I'm closing this, as 1/ a rebase I did on this branch caused a regression in the scrolling, and 2/ it'll be easier if they're divided up into separate PR's anyway. 👍
